compiler/rustc_hir_analysis/src/collect/item_bounds.rs

```diff
@@ -52,6 +139,117 @@ fn associated_type_bounds<'tcx>(
     all_bounds
 }
 
+struct MapAndCompressBoundVars<'tcx> {
```

Comment:
can you add a doc comment here with an example, e.g. 

- `for<'b, 'c> <Self as Trait<'param>>::Gat<'b>: Bound<'c>` results in the mapping `'b` to `'gat_param`
- this folder then converts this to `for<'c> <Self as Trait<'param>>::Gat<'gat_param>: Bound<'c>`